src/robot.coffee

```diff
@@ -75,7 +88,37 @@ class Robot
   #
   # Returns nothing.
   hear: (regex, callback) ->
-    @listeners.push new TextListener(@, regex, callback)
+    @listen(
+      (message) ->
+        if message instanceof TextMessage
+          match = message.match(regex)
+          if match
+            @robot.logger.debug \
+              "Message '#{message}' matched regex /#{inspect regex}/"
+          match
+      callback
+    )
+
+  # Public: Adds a Listener that attempts to match incoming messages based on
+  # a Regex, but rejects all messages that are addressed to the robot
+  # (i.e. anti-respond).
+  #
+  # regex    - A Regex that determines if the callback should be called.
+  # callback - A Function that is called with a Response object.
+  #
+  # Returns nothing.
+  eavesdrop: (regex, callback) ->
```

Comment:
I'm not super happy with the amount of duplication between `eavesdrop` and `hear`. I'm open to suggestions for refactoring.